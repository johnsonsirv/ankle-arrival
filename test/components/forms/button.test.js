/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import Button from '../../../src/components/forms/inputs/button';

function setup() {
  const props = {
    value: 'Send',
    onClick: jest.fn(),
  };
  const enzymeWrapper = mount(<Button {...props} />);

  return { props, enzymeWrapper };
}

describe('<Button /> rendering', () => {
  const { enzymeWrapper } = setup();
  it('renders correctly', () => {
    expect(enzymeWrapper).toMatchSnapshot();
  });
  it('should render 1 editable <button>', () => {
    expect(enzymeWrapper.find('button')).toHaveLength(1);
  });
  it('should set a text value for button', () => {
    expect(enzymeWrapper.find('button').text()).toMatch(/Send/);
  });
  it('should render <button> enabled', () => {
    expect(enzymeWrapper.find('button').prop('disabled')).toBe(false);
  });
});

describe('<Button /> interactions', () => {
  const { props, enzymeWrapper } = setup();
  it('should handle onClick', () => {
    enzymeWrapper.simulate('click', { target: {}, preventDefault: jest.fn() });
    expect(props.onClick).toHaveBeenCalled();
  });
});
