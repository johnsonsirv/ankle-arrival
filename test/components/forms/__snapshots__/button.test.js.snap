// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> rendering renders correctly 1`] = `ReactWrapper {}`;
